week4-ts/my-app/src/components/SearchPage copy.tsx

```diff
@@ -0,0 +1,230 @@
+import React from "react";
+import { useState, useEffect, useRef } from "react";
+import { Outlet } from "react-router-dom";
+import styled from "styled-components";
+
+interface Props {
+  getUser: (username: string) => void;
+}
+
+interface History {
+  id: number,
+  github_id: string,
+}
+
+export default function SearchPage({getUser}: Props) {
+  const [input, setInput] = useState(""); // input 값
+  const [focus, setFocus] = useState(false); // input 포커싱 상태
+  let [histories, setHistories] = useState<History[]>([]); // 검색 히스토리를 저장할 배열
+  const historyStorage = localStorage.getItem("histories"); // 히스토리 로컬스토리지 저장소
+  const nextId = useRef(0);
+
+  useEffect (() => {
+    // initHistory();
+  },[]);
+
+  // 히스토리 로컬스토리지 저장
+  const setLocalStorage = (newHistory: string[]) => {
+    localStorage.setItem("history", JSON.stringify([...histories], newHistory));
+  };
+
+  const addLocalStorage = (newHistory: string) => {
+    const user = {
+      id: nextId.current,
+      github_id: newHistory,
+    };
+    localStorage.setItem("history", JSON.stringify([...histories, user]));
+  }
+
+  const addHistory = (newHistory: string) => {
+    console.log(nextId);
+    const user: History = {
+      id: nextId.current,
+      github_id: newHistory,
+    };
+    setHistories([...histories, user]);
+    nextId.current += 1;
+  }
+
+  // input창 텍스트 보여주기
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => setInput(e.target.value);
+
+  // 검색 클릭 시 history 배열에 넣고 로컬스토리지에 저장
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (input === "") {
+      alert("아이디를 입력하세요");
+    } else {
+      getUser(input);
+      // history 배열에 넣기 (중복 아닐 때)
+      if (histories.filter((history) => history.github_id !== input)) {
+        addHistory(input);
+
+        // 로컬스토리지 업데이트
+        addLocalStorage(input);
+      }
+      setInput("");
+      setFocus(false);
+    }
+  };
+
+  // 히스토리가 새로고침해도 남아있게 하기 위해서 init
+  const initHistory = (): void => {
+    if (historyStorage != null) {
```

Comment:
```suggestion
// 라고만 써도 이 값이 있을때만!! 실행되는 조건문이 됩니당!
if(historyStorage){
```

[javascript null,undefined,empty 체크시 좋은 코딩방법
](https://negabaro.github.io/archive/js-isEmpty)참고!!

Reply:
initHistory 함수와 histoyStorage 배열을 없애고
``` javascript
// 검색 히스토리 저장
const [histories, setHistories] = useState<string[]>(JSON.parse(localStorage.getItem("history") || '[]')); 
```
요렇게 히스토리를 한번에 관리했는데, 요것도 괜찮은 방법일까용?!